### PR TITLE
create private hostnames with port name

### PR DESCRIPTION
**Pull request by matkam**
Hello @dokipen, @ddbenson, @dnelson, @silverlyra, @micahnoland, @davidzhao, 

Please review the commits I made in branch 'matkam/vs-port-names'.

R=@dokipen
R=@ddbenson
R=@dnelson
R=@silverlyra
R=@micahnoland
R=@davidzhao

### CODE REVIEW

**Comment by matkam** [comment]
Verified to create the expected VirtualService in sandbox

File: pkg/controller/releasemanager/plan/syncApp.go [hunk 1]
```diff
@@ -342,15 +316,42 @@ func (p *SyncApp) virtualService(log logr.Logger) *istiov1alpha3.VirtualService
 		return nil
 	}
 
+	hostsMap := map[string]bool{
```
Comment (silverlyra) [inline pkg/controller/releasemanager/plan/syncApp.go]: why were these inlined?

File: pkg/controller/releasemanager/plan/syncApp.go [hunk 2]
```diff
@@ -342,15 +316,42 @@ func (p *SyncApp) virtualService(log logr.Logger) *istiov1alpha3.VirtualService
...
+			}
+			hostsMap[httpMatch.Authority.Prefix] = true
+		}
+	}
+	hosts := make([]string, len(hostsMap))
+	var i int
+	for host := range hostsMap {
+		hosts[i] = host
+		i++
+	}
```
Comment (silverlyra) [inline pkg/controller/releasemanager/plan/syncApp.go]:
```suggestion
	hosts := make([]string, 0, len(hostsMap))
	for host := range hostsMap {
		hosts = append(hosts, host)
	}
```
save the explicit index management

File: pkg/controller/releasemanager/plan/syncApp.go [hunk 3]
```diff
@@ -126,7 +123,14 @@ func (p *SyncApp) releaseMatches(log logr.Logger, port picchuv1alpha1.PortInfo)
...
 		}
 		gateway = []string{p.PrivateGateway}
-		hosts = append(hosts, p.defaultHost())
+		hosts = append(hosts, fmt.Sprintf("%s.%s", p.Namespace, p.DefaultDomain)) // default host
+
+		portName := port.Name
+		if portName == "" {
+			portName = strconv.Itoa(int(port.Port))
+		}
+		hosts = append(hosts, fmt.Sprintf("%s-%s.%s", p.Namespace, portName, p.DefaultDomain))
```
Comment (silverlyra) [inline pkg/controller/releasemanager/plan/syncApp.go]:
```suggestion
		if port.Name != "" {
			hosts = append(hosts, fmt.Sprintf("%s-%s.%s", p.Namespace, port.Name, p.DefaultDomain))
		}
```
how about only supporting this for named ports?

File: pkg/controller/releasemanager/plan/syncApp.go [hunk 4]
```diff
@@ -4,6 +4,7 @@ import (
 	"context"
 	"fmt"
 	"sort"
+	"strconv"
```
Comment (silverlyra) [inline pkg/controller/releasemanager/plan/syncApp.go]:
```suggestion
```

**Pull request merged**